### ISSUE TITLE
Preliminary demand estimates with NC scratch-off data

### ISSUE THREAD

**Issue by huntallcott** [huntallcott/Lotteries#9]
@jackblun writes:
Hi Hunt,

Just to check we are on the same page - the basic idea with the scratch-off data is as follows?
1) Approximate monthly demand by looking at number of sales to date
2) Relate demand to characteristics of scratch-offs

If so, I can start on the code to do this and we can discuss on Tuesday.

**
HA responds:
Let me clarify a bit on your 1 and 2 above:
1) Approximate monthly quantity demanded; call this s_j. You should be able to approximate total quantity planned for the game using the "overall odds" and the counts of each prize level. Then you approximate the share of total planned quantity that's been sold to date by comparing "prize count" to "remaining count." Then you get sales per month by looking at the time between "launch date" and {now or "end date"}. I presume that this is what you had in mind but just checking.
2) Specifically, I would go ahead and try to estimate the logit model we laid out. For now, you should assume all consumers are homogeneous, so all consumers' logit choice probabilities are the same, and you get the estimating equation:
ln(s_j) - ln(s_0) = V_j,
where s_0 is the outside option share and V_j is the indirect utility from game j.
(I presume that you know that this can be derived by assuming homogeneous consumers and re-arranging Equation 5 from DemandModel.pdf.)

We can make some assumption about s_0 (outside option share), e.g. using total market size where there are 10 million people in NC and each has one choice occasion per week. (Although it won't matter what you assume for s_0 because that comes out in a constant.) 

Then get the V_j from the functional form assumptions like those in DemandModel.pdf, i.e. w(\pi) is as in equation 1 (or whatever you want) and u(x;\rho) is CARA. (I had written CRRA in the pdf, but @bblockwood says he might want CARA.)

In other words, I'm saying to estimate equation (6) in DemandModel.pdf, except that you have ln(s_j/s_0) instead of \delta_{st} as the dependent variable. You should have 75 observations in the dataset (one for each scratch-off game) and should be able to estimate with non-linear least squares in Stata.

Does this make sense? Thanks for getting this going.

**Comment by jackblun** [comment]
@huntallcott @bblockwood @dtaubinsky  I've done some estimation, pushed the code and written up thoughts [here](https://docs.google.com/document/d/1Iw7GFYAGwhqU-RN7F-kFpHuS1i1McYLFwsT8bMO2Qbo/edit#).

As you'll see, the data is reasonable and the results are very unstable - I'm thinking there is an issue with identification, conditional on my non linear least squares code being correct. I should caveat that this is the first time I've used the nl command in stata. The code should be fairly self explanatory if anyone fancies having a look. It will be good to discuss what we think is going on later.

**Comment by huntallcott** [comment]
@jackblun thanks for doing this.
Seems like there are three possibilities:
1. alpha and rho are not identified
2. alpha and rho are identified but we don't have great variation in the data
3. there's something wrong with the code

### One preliminary comment
You do need an alternative-specific constant \xi_g in the V_{ig} equation. I added a quick comment in the google doc to that effect. Then since \xi_g is not necessarily mean-zero, that's how you get both a constant term and the error terms in the actual estimation. For that reason, please estimate a constant, although it seems like that's not the problem.

### Diagnosing code errors
Here is one way to diagnose whether (3) is the issue: what if we simplify the problem so it's linear, then estimate the market share equation in both OLS and NLS. One way to make it linear is as follows:
a. Assume log utility, so u(x) = ln(x+B), where B>p so that you never have log of a non-positive number. It may matter for the estimates what you assume for B, but this is just a test.
b. Assume @bblockwood 's sort-of linear probability weighting function, so w(\pi;alpha0,alpha1) = alpha0+alpha1*p, for p>0, and w=0, for p=0.

Then you get V_ig = an additively separable function of alpha1 and alpha0, and you can estimate alpha1 and alpha0 in OLS AND also NLS and make sure that they give the same results.


### Simulated sales data.
Another way to do this is *simulate* scratch-off sales data for each of the 75 scratch-offs using an assumed alpha and rho.  This will give a set of market shares. Then plug those market shares into the estimation and see if you get back your original alpha and rho. This will tell us both if the estimation code is working and whether alpha and rho are in principle identified.


I hope that this makes sense and is useful. Thanks again for working on it!

**Comment by huntallcott** [comment]
@jackblun, upon reflection, I would do the "simulated sales data" thing above before the "diagnosing code errors." 
On that, I'm also curious what the simulated data look like and if they have some different variation than the actual sales data.
Or it could be that we need the game characteristics to have more variation, e.g. you have shown that the tickets all have similar expected value conditional on the price, and maybe that hurts identification.

**Comment by jackblun** [comment]
Thanks for all this. Agreed that lack of identification is only one of three potential explanations - apologies I couldn't dive in further today. A couple of responses:
1) Re the constant, noted. I tried models with and without and got broadly similar results. 
2) In expected value terms, you're right that there is not much variation in the data. However there is a huge amount of variation in odds and outcomes, which I would have thought would be good for us.

We can discuss in the call but I think the simulation approach sounds like the way to go.

**Comment by jackblun** [comment]
@huntallcott I've just spent the day on this. I found one coding error in the way I defined the outcome variable, but it doesn't help things massively. 

I've set up a lot of simulations (see the code if interested) and am getting similar problems in the simulated data. 

Do you know of any alternative method in stata rather than the nl package for this kind of thing?

**Comment by huntallcott** [comment]
@jackblun thanks for working on this.
Two thoughts:
1. Stata should be OK on small optimization problems. I wonder if we just don't have the right variation in characteristics of the tickets. When you did the simulations, did you simulate just the market shares, or did you also simulate different scratch-off tickets with different characteristics? What if you, e.g., simulate other scratch-offs with characteristics that should definitely work to identify curvature and non-linear probability weighting, and then simulate the market shares of those simulated scratch-offs, and then estimate curvature and weighting from that? e.g. just a small number of scratch-offs that have the same EV but VERY different jackpots, to get curvature, and another few scratch-offs that vary only moderate probabilities and high probabilities, to get the weighting function.
2. If Stata really isn't working, then R or MATLAB would be natural, and perhaps easier than writing NLS in Stata because their matrix notation is more standard.
Thanks again.

**Comment by jackblun** [comment]
@huntallcott I've got the simulations working. The problem was partly with stata's nlls command being vulnerable to finding local optima. By experimenting with starting values for the optimization, I think I got to the bottom of it. 

In a nutshell:
1) We can identify rho and alpha from market shares
2) The NC data we have leads to a very imprecise estimate of alpha -> more data should help

I wrote up what I found [here](https://docs.google.com/document/d/1Iw7GFYAGwhqU-RN7F-kFpHuS1i1McYLFwsT8bMO2Qbo/edit#). All code pushed.
